webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -28,10 +28,37 @@ import { MessageDuration } from "../message-duration";
 import * as monaco from 'monaco-editor'
 
 let stateCache = {};
+const WHITESPACE_256 = String.fromCharCode.apply(null, new Array(256).fill(0x20));
 let lastFile;
 //Unsupported DS types
 const unsupportedTypes: Array<string> = ['G', 'B', 'C', 'D', 'I', 'R'];
 
+function isContentValidForDataset(content: string[], datasetAttrs: DatasetAttributes) {
+  //TODO: validation of record length that is aware of how DBCS will effect actual length
+  //FB must have exactly lrecl, VB must have no more than lrecl. content cant be undefined.
+  if (!content) {
+    return 'No Content';
+  }
+  let isFixedRecordLength = datasetAttrs.recfm.recordLength === 'F';
+  let maxRecordLen = datasetAttrs.dsorg.maxRecordLen;
+  for (let i = 0; i < content.length; i++) {
+    let record = content[i];
+    if (record.length > maxRecordLen) {
+      return `Line ${i} exceeds record length limit of ${maxRecordLen}`;
+    } else if (isFixedRecordLength && record.length < maxRecordLen) {
+      let whitespace = WHITESPACE_256;
+      while (whitespace.length < maxRecordLen) {
+        whitespace += WHITESPACE_256;
+      }
+      content[i] = record+whitespace.substring(record.length,maxRecordLen);
+    }
+  }
+  if (JSON.stringify({records:content}).length > 5242880) {
+    return 'Content over max size currently supported (5MB)';
```

Comment:
And this should be a (CONST / 1024) for bytes or (CONST / (1024 * 1024) for MB

(no hardcoding)

Reply:
In the future, say the Editor "Settings" menu, we may want to have this value customizable & not CONST